### DISCUSSION TITLE
When starting the .devcontainer configuration where can one find the log that logs how mastodon starts up?

### DISCUSSION THREAD

**Discussion by ohthatwright** [mastodon/mastodon#21596] [Q&A]
I'm starting up the .devcontainer configuration (Windows 11, VSCode remote, switching to Dev Container). I'm not getting any signs of life when I go to port 3000 (I've tried various ways).  Would like to go examine Mastodon's start uplogs and see if I can find what might be failing, but don't know where to look. Realizing that prior Rails experience would really help, but I don't have it. Thank you!

**Comment by jaykz52** [comment]
@ohthatwright FYI I've been poking around at this for a few hours today, and here's what worked for me (based off of `main` as of this morning):

1. Start devcontainer (e.g. using VSCode)
2. Run `$ docker ps` and get the container ID of the `mastodon_devcontainer_app` image
3. `$ docker exec -it CONTAINER_ID bash`
4. `$ cd mastodon`
5. `$ RAILS_ENV=development ./bin/rails db:setup` to generate the `mastodon_development` DB
6. `$ foreman start`
7. Wait for the various processes to start...
8. Visit `localhost:3000` from a web browser

I'm not sure yet how to incorporate this into the existing setup, but wanted to share in case it helps you or others.

**Reply by ohthatwright** [reply]
Ty! Am in transit but will try your approach soon. Really appreciate it.

**Reply by jaykz52** [reply]
Great. Please do let me know if you have success - I'd like to get the Dev Container setup into a more turn-key state for others going forward.

**Comment by astridx** [comment]
@jaykz52  Is it possible for you to exchange messages with other computers that are also installed locally (follow, nodes)?

I ask because I am stuck with a local installation at this point. I think it is because localhost:3000 is not encrypted (no https).

https://github.com/mastodon/mastodon/discussions/22309

https://discourse.joinmastodon.org/t/local-deployment/1018
